### PR TITLE
Methods do not have caller and arguments properties

### LINKED ISSUE #4655: Object member functions handle caller/arguments incorrectly.
Follow-up #4651

/cc @boingoing

### PR DESCRIPTION
**Pull request by boingoing**
We shouldn't be creating caller and arguments properties for object-literal methods.

See: https://tc39.github.io/ecma262/#sec-forbidden-extensions

Fixes #4655

### CODE REVIEW

File: test/es6/ES6RestrictedProperties.js [hunk 1]
```diff
@@ -369,7 +364,7 @@ var tests = [
         body: function () {
             var obj = { func() { } }
```
Comment (agarwal-sandeep) [inline test/es6/ES6RestrictedProperties.js]: >var obj = { func() { } } [](start = 12, length = 24)

Do we need to have cases like

var obj = { func:function() { } }
function foo() {}
var obj = { func:foo }
var obj = { func:Math.sin }

Comment (boingoing) [reply test/es6/ES6RestrictedProperties.js]: These cases are not methods - they should behave like ordinary functions.

Comment (agarwal-sandeep) [reply test/es6/ES6RestrictedProperties.js]: Good to add test to verify that?

---
In reply to: [174215572](https://github.com/Microsoft/ChakraCore/pull/4810#discussion_r174215572) [](ancestors = 174215572)

**Comment by dilijev** [comment]
Thanks for this, @boingoing!

**Pull request merged**